providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python3
+# This file is part of Checkbox.
+#
+# Copyright 2023 Canonical Ltd.
+# Written by:
+#   Hanhsuan Lee <hanhsuan.lee@canonical.com>
+#
+# Checkbox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# Checkbox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Checkbox.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import threading
+import subprocess
+import time
+import re
+import json
+import argparse
+import logging
+import os
+
+
+class PrimeOffloader:
+    """
+    A class used to execute process to specific GPU.
+    Have to run this as root.
+
+    :attr logger: console logger
+    :type logger: obj
+
+    :attr check_result:
+        store the result of checking offloading is ok or not.
+    :type check_result: int
+    """
+
+    logger = logging.getLogger()
+    check_result = False
+
+    def find_card_id(self, pci_name):
+        """
+        use pci name to find card id under /sys/kernel/debug/dri
+
+        :param pci_name: pci device name in NNNN:NN:NN.N format
+        :type card_name: str
+
+        : returns : card id
+        : rtype : str
+        """
+        if not re.match("[0-9]{4}:[0-9]{2}:[0-9]{2}.[0-9]", pci_name):
+            raise RuntimeError("pci name format error")
```

Comment:
Do we want the whole test to crash with a backtrace? Or do we want to exit the test program with a tidy error message and an error code? 
If it's the latter then I recommend raising `SystemExit`.